src/Xamarin.Android.Build.Tasks/Utilities/TypeMapGenerator.cs

```diff
@@ -329,12 +377,18 @@ bool GenerateRelease (bool skipJniAddNativeMethodRegistrationAttributeScan, List
 				}
 
 				string javaName = Java.Interop.Tools.TypeNameMappings.JavaNativeTypeManager.ToJniName (td);
+				// We will ignore generic types and interfaces when generating the Java to Managed map, but we must not
+				// omit them from the table we output - we need the same number of entries in both java-to-managed and
+				// managed-to-java tables.  `SkipInJavaToManaged` set to `true` will cause the native assembly generator
+				// to output `0` as the token id for the type, thus effectively causing the runtime unable to match such
+				// a Java type name to a managed type. This fixes https://github.com/xamarin/xamarin-android/issues/4660
 				var entry = new TypeMapReleaseEntry {
 					JavaName = javaName,
 					JavaNameLength = outputEncoding.GetByteCount (javaName),
 					ManagedTypeName = td.FullName,
 					Token = td.MetadataToken.ToUInt32 (),
-					AssemblyNameIndex = knownAssemblies [assemblyName]
+					AssemblyNameIndex = knownAssemblies [assemblyName],
+					SkipInJavaToManaged = td.IsInterface | td.HasGenericParameters,
```

Comment:
Shouldn't this be `||`, not `|`?

Reply:
it can be either, since it's a `bool` it will work both ways, but I guess `||` would be more conventional...